src/sonar_widget/SonarPlot.cc

```diff
@@ -76,23 +85,24 @@ bool SonarPlot::isMotorStepChanged(const base::Angle& bearing) {
     base::Angle diffStep = bearing - lastSonar.bearings[0];
     diffStep.rad = fabs(diffStep.rad);
 
-    if (!motorStep.isApprox(diffStep)) {
+    if (!motorStep.isApprox(diffStep) && lastDiffStep.isApprox(diffStep) && !bearing.isApprox(beforeLastSonar.bearings[0])) {
         motorStep = diffStep;
         numSteps = round(M_PI * 2 / motorStep.rad);
```

Comment:
`round()` looks wrong here. I don't think it makes sense to have `numSteps * motorStep.rad > PI*2`